### PR TITLE
DVCSMP-2324 - Centralite Thermostat handle multiple attributes

### PR DESCRIPTION
**Pull request by juano2310**

### CODE REVIEW

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy [hunk 1]
```diff
@@ -109,13 +114,10 @@ def parse(String description) {
 			map.name = "thermostatFanMode"
 			map.value = getFanModeMap()[descMap.value]
 		}
+		if (map)
```
Comment (varzac) [inline devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: I prefer to be explicit and use curly braces even on one line ifs.

**Review by varzac — CHANGES_REQUESTED** [review]
You can delete the parseDescriptionAsMap included in this DTH as it is no longer used.

**Comment by varzac** [comment]
I talked with Tom, and I think we are going to add changes to the parseDescriptionAsMap to automatically put all the attr data together, but I will make changes to the DTH when that is supported.

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy (hunk 1 above)
Comment (juano2310) [reply devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: Done :)

**Comment by juano2310** [comment]
parseDescriptionAsMap removed :)

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy [hunk 2]
```diff
@@ -81,10 +81,15 @@ metadata {
...
+	def result = null
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "Desc Map: $descMap"
+	List attrData = [[cluster: descMap.cluster ,attrId: descMap.attrId, value: descMap.value]]
+	descMap.additionalAttrs.each {
+	    attrData << [cluster: descMap.cluster, attrId: it.attrId, value: it.value]
+	}
+	attrData.each {
+		def map = [:]
 		if (descMap.cluster == "0201" && descMap.attrId == "0000") {
```
Comment (tpmanley) [inline devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: Need to change `descMap` references to `it`

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy [hunk 3]
```diff
@@ -81,10 +81,15 @@ metadata {
 // parse events into attributes
 def parse(String description) {
 	log.debug "Parse description $description"
-	def map = [:]
-	if (description?.startsWith("read attr -")) {
-		def descMap = parseDescriptionAsMap(description)
-		log.debug "Desc Map: $descMap"
+	def result = null
```
Comment (tpmanley) [inline devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: This should probably be made into a list and events appended to it

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy [hunk 4]
```diff
@@ -109,23 +114,14 @@ def parse(String description) {
 			map.name = "thermostatFanMode"
 			map.value = getFanModeMap()[descMap.value]
 		}
+		if (map) {
+			result = createEvent(map)
```
Comment (tpmanley) [inline devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: The events should be appended to the `result` list

Comment (juano2310) [reply devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: done

File: devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy [hunk 5]
```diff
@@ -81,51 +81,47 @@ metadata {
 // parse events into attributes
 def parse(String description) {
 	log.debug "Parse description $description"
-	def map = [:]
-	if (description?.startsWith("read attr -")) {
-		def descMap = parseDescriptionAsMap(description)
-		log.debug "Desc Map: $descMap"
-		if (descMap.cluster == "0201" && descMap.attrId == "0000") {
+	List result = null
```
Comment (tpmanley) [inline devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: Needs to be initialized to a list

Comment (juano2310) [reply devicetypes/smartthings/centralite-thermostat.src/centralite-thermostat.groovy]: done :)

**Review by tpmanley — APPROVED** [review]

**Pull request merged**